packages/siaviewnode-client/src/components/Dashboard.tsx

```diff
@@ -0,0 +1,161 @@
+                {props.title}
+            </span>
+        </div>
+        <FontAwesomeIcon icon={props.icon} size={"2x"} transform={props.iconTransform} style={{ color: theme.colors.primary }}/> 
+        <span style={{ fontSize: "32px", fontWeight: "bold", paddingLeft: "15px", color: theme.colors.text }}>{props.stat}
+            <span style={{fontSize: "50%", paddingLeft: 5}}>
+                {props.suffix}
+            </span>
+        </span>
+    </Col>
+}
+
+// Custom component: StatBarChart
+function StatBarChart(props) {
+    // "series" is an object containing the name of the data serie and the color for the chart. It is
+    // mapped to a <Bar> component. Its format as an input prop is, as an example:
+    // {"p80": "#8884d8", "p95": "#82ca9d", "p99": "#666"}
+    const series = props.series
+        
+    return (
+        <div style={{ paddingTop: 40, color: theme.colors.text }}>
+            <FontAwesomeIcon icon={props.icon} />
+            <span style={{ fontSize: "16px", fontWeight: "bold", paddingLeft: "10px" }}>{props.title}</span>
+            <div style={{ height: 350 }}>
+                <ResponsiveContainer>
+                    <BarChart
+                        data={props.data}
+                        margin={{
+                        top: 25, right: 25, left: 25, bottom: 5,
+                        }}
+                    >
+                        <CartesianGrid strokeDasharray="3 3" />
+                        <XAxis dataKey="name" stroke={ theme.colors.text }/>
+                        <YAxis stroke={ theme.colors.text } label={
+                            <Text fill={theme.colors.text} dx={30} dy={180} offset={0} angle={-90}>{props.units}</Text> 
+                        }/>
+                        <Tooltip  />
+                        <Legend />
+                        {Object.keys(series).map(function(key, index) {
+                            return <Bar dataKey={key} fill={series[key]} />
+                        })}
+                    </BarChart>
+                </ResponsiveContainer>
+            </div>
+        </div>
+    )
+}
+
+function colorChanger(hex, rgbAddition) {
+    // For the charts we create variations of the primary color
+    // This function transforms the HEX value into rgb, adds or substracts "rgbAddition"
+    // to eeach color component and transforms again into a new HEX value. By default, I 
+    // recommend a value of "rgbAddotion" = 50
+    try {
+        var rgb = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
+        var r = parseInt(rgb[1], 16) + rgbAddition
+        if (r > 255) {r = 255}
+        if (r < 0) {r = 0}
+        var g = parseInt(rgb[2], 16) + rgbAddition
+        if (g > 255) {g = 255}
+        if (g < 0) {g = 0}
+        var b = parseInt(rgb[3], 16) + rgbAddition
+        if (b > 255) {b = 255}
+        if (b < 0) {b = 0}
```

Comment:
This works well and can definitely stay as it is. If we want to DRY it up a little we can try something like this as well:
```
const normalise = (x) => {
  if (x < 0) { return 0 }
  if (x > 255) { return 255 }
  return x
}
let r = normalise(parseInt(rgb[1], 16) + rgbAddition)
...
```